test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala

```diff
@@ -81,536 +82,367 @@ class AddBusinessNameControllerSpec extends TestSupport
-      "user has already completed the journey" in {
-        disableAllSwitches()
-        enable(IncomeSources)
-
-        mockNoIncomeSources()
-        setupMockAuthRetrievalSuccess(BaseTestConstants.testIndividualAuthSuccessWithSaUtrResponse())
-        setupMockGetMongo(Right(Some(sessionDataCompletedJourney)))
-
-        val result: Future[Result] = TestAddBusinessNameController.changeBusinessName()(fakeRequestWithActiveSession)
-        status(result) shouldBe SEE_OTHER
-        val redirectUrl = controllers.incomeSources.add.routes.ReportingMethodSetBackErrorController.show(SelfEmployment).url
-        redirectLocation(result) shouldBe Some(redirectUrl)
-      }
-      "user has already added their income source" in {
-        disableAllSwitches()
-        enable(IncomeSources)
-
-        mockNoIncomeSources()
-        setupMockAuthRetrievalSuccess(BaseTestConstants.testIndividualAuthSuccessWithSaUtrResponse())
-        setupMockGetMongo(Right(Some(sessionDataISAdded)))
-
-        val result: Future[Result] = TestAddBusinessNameController.changeBusinessName()(fakeRequestWithActiveSession)
-        status(result) shouldBe SEE_OTHER
-        val redirectUrl = controllers.incomeSources.add.routes.IncomeSourceAddedBackErrorController.show(SelfEmployment).url
-        redirectLocation(result) shouldBe Some(redirectUrl)
-      }
-    }
-  }
 
-  "Individual - Change - AddBusinessNameController.submitChange" should {
-    "change business name" when {
-      "redirect to the check business details" when {
-        "the individual is authenticated" in {
-          disableAllSwitches()
-          enable(IncomeSources)
-
-          setupMockAuthRetrievalSuccess(BaseTestConstants.testIndividualAuthSuccessWithSaUtrResponse())
-          setupMockGetIncomeSourceDetails()(businessesAndPropertyIncome)
-          setupMockSetSessionKeyMongo(businessNameField, validBusinessName, journeyType, Right(true))
-          setupMockGetSessionKeyMongoTyped[String](businessNameField, journeyType, Right(Some(validBusinessName)))
-          setupMockGetSessionKeyMongoTyped[String](businessTradeField, journeyType, Right(Some("Test Business Trade")))
-
-          val redirectUrl = controllers.incomeSources.add.routes.IncomeSourceCheckDetailsController.show(SelfEmployment).url
-
-          val result: Future[Result] = TestAddBusinessNameController.submitChange()(fakePostRequestConfirmedClient().withFormUrlEncodedBody(
-            BusinessNameForm.businessName -> validBusinessName))
-          status(result) shouldBe SEE_OTHER
-          redirectLocation(result) mustBe Some(redirectUrl)
-          verify(mockSessionService)
-            .setMongoKey(ArgumentMatchers.eq(businessNameField), ArgumentMatchers.eq(validBusinessName), ArgumentMatchers.eq(journeyType))(any(), any())
+      s"ADD - AddBusinessNameController.submit (${if (isAgent) "Agent" else "Individual"}, isChange = $isChange)" should {
+        "return 303 and redirect to add business start date" when {
```

Comment:
❓ Are you testing all scenarios for isAgent and isChange here? I can't spot where you're running through the inputs

Reply:
There's a 
for (isAgent <- Seq(true, false)) yield {
    for (isChange <- Seq(true, false)) yield {
around the whole thing, so it essentially runs the whole thing 4 times with the different scenarios